pkg/roachprod/failureinjection/failures/network_partition.go

```diff
@@ -0,0 +1,239 @@
+	addOrDropRule := "-D"
+	if addRule {
+		addOrDropRule = "-A"
+	}
+	return fmt.Sprintf(partitionTemplateWrapper, fmt.Sprintf(partitionCmd, addOrDropRule, targetNode))
+}
+
+func (f *IPTablesPartitionNode) Inject(
+	ctx context.Context, l *logger.Logger, args FailureArgs,
+) error {
+	partitions := args.(NetworkPartitionArgs).Partitions
+	for _, partition := range partitions {
+		for _, destinationNode := range partition.Destination {
+			var cmd string
+			switch partition.Type {
+			case Bidirectional:
+				cmd = constructIPTablesRule(bidirectionalPartitionCmd, destinationNode, true /* addRule */)
+				l.Printf("Dropping packets between nodes %d and node %d with cmd: %s", partition.Source, destinationNode, cmd)
+			case Incoming:
+				cmd = constructIPTablesRule(asymmetricInputPartitionCmd, destinationNode, true /* addRule */)
+				l.Printf("Dropping packets from node %d to nodes %d with cmd: %s", destinationNode, partition.Source, cmd)
+			case Outgoing:
+				cmd = constructIPTablesRule(asymmetricOutputPartitionCmd, destinationNode, true /* addRule */)
+				l.Printf("Dropping packets from nodes %d to node %d with cmd: %s", partition.Source, destinationNode, cmd)
+			default:
+				panic("unhandled default case")
+			}
+			if err := f.run(ctx, l, partition.Source, cmd); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (f *IPTablesPartitionNode) Restore(
+	ctx context.Context, l *logger.Logger, args FailureArgs,
+) error {
+	partitions := args.(NetworkPartitionArgs).Partitions
+	for _, partition := range partitions {
+		for _, destinationNode := range partition.Destination {
+			var cmd string
+			switch partition.Type {
+			case Bidirectional:
+				cmd = constructIPTablesRule(bidirectionalPartitionCmd, destinationNode, false /* addRule */)
+				l.Printf("Resuming packets between nodes %d and node %d with cmd: %s", partition.Source, destinationNode, cmd)
+			case Incoming:
+				cmd = constructIPTablesRule(asymmetricInputPartitionCmd, destinationNode, false /* addRule */)
+				l.Printf("Resuming packets from node %d to nodes %d with cmd: %s", destinationNode, partition.Source, cmd)
+			case Outgoing:
+				cmd = constructIPTablesRule(asymmetricOutputPartitionCmd, destinationNode, false /* addRule */)
+				l.Printf("Resuming packets from nodes %d to node %d with cmd: %s", partition.Source, destinationNode, cmd)
+			default:
+				panic("unhandled default case")
+			}
+			if err := f.run(ctx, l, partition.Source, cmd); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (f *IPTablesPartitionNode) Cleanup(_ context.Context, _ *logger.Logger, _ FailureArgs) error {
+	return nil
+}
+
+// PacketsDropped returns the number of packets dropped on a given node due to iptables rules.
+func (f *IPTablesPartitionNode) PacketsDropped(
+	ctx context.Context, l *logger.Logger, node install.Nodes,
+) (int, error) {
+	res, err := f.runWithDetails(ctx, l, node, "sudo iptables -L -v -n")
```

Comment:
Maybe add `-x` to get exact numbers; else in case of a lot of dropped packets, the output will be rounded to the nearest K/M/G (e.g. `1104K` instead of `1104123`).

Reply:
We should ideally grab only the network interface that's being exercised. Otherwise, there is a potential to introduce noise.

Reply:
Both suggestions SGTM, but I actually just removed the function entirely for now. That was originally added in the `network.go` tests as a check to make sure the rules were actually applied. Once the test switches to the failure injection library, it shouldn't have to own that validation anymore. Will add it back in if needed.